src/blueprints/TownHall.cpp

```diff
@@ -7,9 +7,12 @@
 #include "core/API.h"
 
 bool TownHall::Build(Order* order_) {
-    const sc2::Point3D* town_hall_location = gHub->GetNextExpansion();
-    if (!town_hall_location)
+    Expansion* next_expand = gHub->GetNextExpansion();
+    if (!next_expand)
         return false;
 
-    return gHub->AssignBuildTask(order_, *town_hall_location);
+    sc2::Tag builder = gHub->AssignBuildTask(order_, next_expand->town_hall_location);
+    next_expand->worker_tag = builder;
```

Comment:
Now it looks a bit weird as `AssignBuildTask` can return zero, but which means that we can't fulfil the order, the the actual return value is not checked.

Reply:
Ok, made logic explicit.